doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -126,114 +152,156 @@ Type declarations
+   :name: Ltac2 external
 
-      Open variants are a special kind of variant types whose constructors are not
-      statically defined, but can instead be extended dynamically. A typical example
-      is the standard `exn` type. Pattern matching on open variants must always include a catch-all
-      clause. They can be extended with this command.
+   Declares abstract terms.  Frequently, these declare OCaml functions
+   defined in |Coq| and give their type information.  They can also declare
+   data structures from OCaml.  It has no use for the end user.
+
+APIs
+~~~~
+
+Ltac2 provides over 150 API functions that provide various capabilities.  These
+are declared with :cmd:`Ltac2 external` in :n:`lib/coq/user-contrib/Ltac2/*.v`.
+For example, `Message.print` defined in `Message.v` is used to print messages:
+
+.. coqtop:: none
+
+   Goal True.
+
+.. coqtop:: all abort
+
+   Message.print (Message.of_string "fully qualified calls").
+   From Ltac2 Require Import Message.
+   print (of_string "unqualified calls").
 
 Term Syntax
 ~~~~~~~~~~~
 
-The syntax of the functional fragment is very close to the one of Ltac1, except
+The syntax of the functional fragment is very close to that of Ltac1, except
 that it adds a true pattern-matching feature, as well as a few standard
 constructs from ML.
 
-.. productionlist:: coq
-   ltac2_var        : `lident`
-   ltac2_qualid     : ( `modpath` . )* `lident`
-   ltac2_constructor: `uident`
-   ltac2_term       : `ltac2_qualid`
-                    : `ltac2_constructor`
-                    : `ltac2_term` `ltac2_term` ... `ltac2_term`
-                    : fun `ltac2_var` => `ltac2_term`
-                    : let `ltac2_var` := `ltac2_term` in `ltac2_term`
-                    : let rec `ltac2_var` := `ltac2_term` in `ltac2_term`
-                    : match `ltac2_term` with `ltac2_branch` ... `ltac2_branch` end
-                    : `int`
-                    : `string`
-                    : `ltac2_term` ; `ltac2_term`
-                    : [| `ltac2_term` ; ... ; `ltac2_term` |]
-                    : ( `ltac2_term` , ... , `ltac2_term` )
-                    : { `ltac2_field` `ltac2_field` ... `ltac2_field` }
-                    : `ltac2_term` . ( `ltac2_qualid` )
-                    : `ltac2_term` . ( `ltac2_qualid` ) := `ltac2_term`
-                    : [; `ltac2_term` ; ... ; `ltac2_term` ]
-                    : `ltac2_term` :: `ltac2_term`
-                    : ...
-   ltac2_branch     : `ltac2_pattern` => `ltac2_term`
-   ltac2_pattern    : `ltac2_var`
-                    : _
-                    : ( `ltac2_pattern` , ... , `ltac2_pattern` )
-                    : `ltac2_constructor` `ltac2_pattern` ... `ltac2_pattern`
-                    : [ ]
-                    : `ltac2_pattern` :: `ltac2_pattern`
-   ltac2_field      : `ltac2_qualid` := `ltac2_term`
-
 In practice, there is some additional syntactic sugar that allows e.g. to
```

Comment:
```suggestion
In practice, there is some additional syntactic sugar that allows the user to e.g.
```

Reply:
I think this correction is needed for the sentence to have correct grammar, so I'm unresolving it, as the change doesn't seem to have been integrated

Reply:
Already fixed in the "More changes commit".